### ISSUE TITLE
Flutter Engine should consider banning static initializers

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#18588]
See Chromium's policy and reasoning:
https://chromium.googlesource.com/chromium/src/+/lkcr/docs/static_initializers.md
http://neugierig.org/software/chromium/notes/2011/08/static-initializers.html

Unless maybe already our build rules do?  @chinmaygarde might know.

**Comment by chinmaygarde** [comment]
No. Our current build rules do not ban static initializers. Though, I have tried addding the same in the past. There are a few dependencies that do not compile with that updated options. I think we should ban them in engine code to start and then expand scope from there.

**Comment by pedromassangocode** [comment]
Is this still a valid issue?

**Comment by chinmaygarde** [comment]
Yes. We have fewer globals and singletons in the engine than when this issue was first written. But getting rid of a few (especially some in our dependencies) is hard. Maybe we can consider a ban for additions.
